frame/support/src/inherent.rs

```diff
@@ -64,21 +69,30 @@ pub trait ProvideInherent {
 	/// included in the block by its author. Whereas the second parameter represents the inherent
 	/// data that the verifying node calculates.
 	///
-	/// NOTE: A block can contains multiple inherent.
+	/// This is intended to allow for checks that cannot be done within the runtime such as, e.g.,
+	/// the timestamp.
+	///
+	/// NOTE: A block can contain multiple inherents.
+	///
+	/// # Warning
+	///
+	/// This check is not guaranteed to be run by all full nodes and cannot be relied upon for
+	/// ensuring correct block import. Run security relevant checks in the extrinsic returned by
+	/// [`Self::create_inherent`] as much as possible.
```

Comment:
```suggestion
	/// This check is not guaranteed to be run by all full nodes and cannot be relied upon for
	/// ensuring that the block is correct.
```

For the second sentence, you already say above that this is only for checking stuff that can not be checked from within the runtime